### PR TITLE
Docs: remove some lies from cross referencing guide

### PR DESCRIPTION
**Pull request by stsewd**
closes https://github.com/readthedocs/readthedocs.org/issues/8172

### CODE REVIEW

File: docs/guides/cross-referencing-with-sphinx.rst [hunk 1]
```diff
@@ -77,11 +77,6 @@ Another example, here we add a target to a paragraph:
 Then we can reference it using ```target to paragraph`_``,
 that will be rendered as: `target to paragraph`_.
 
-The reference displays the name of the target by default,
-but we can use any text you want. For example:
-```my custom text <target to paragraph>`_``,
-that will be rendered as: `my custom text <target to paragraph>`_.
```
Comment (astrojuanlu) [inline docs/guides/cross-referencing-with-sphinx.rst:83]: Why is this a lie? 🤔

Comment (stsewd) [reply docs/guides/cross-referencing-with-sphinx.rst:83]: you can't do this, it isn't supported to set a custom text.

Comment (stsewd) [reply docs/guides/cross-referencing-with-sphinx.rst:83]: yeah, but the generated link is wrong, it should be https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#target-to-paragraph

Comment (astrojuanlu) [reply docs/guides/cross-referencing-with-sphinx.rst:83]: Oh wow, that was subtle. So much reST to learn still...

**Review by astrojuanlu — APPROVED** [review]

**Pull request merged**